wdl-format/src/v1.rs

```diff
@@ -0,0 +1,309 @@
+use crate::Writable as _;
+use crate::element::FormatElement;
+
+/// Formats an [`Ast`](wdl_ast::Ast).
+pub fn format_ast(element: &FormatElement, stream: &mut TokenStream<PreToken>) {
+    let mut children = element.children().expect("AST children");
+
+    let version_statement = children.next().expect("version statement");
+    assert!(version_statement.element().kind() == SyntaxKind::VersionStatementNode);
+    (&version_statement).write(stream);
+
+    stream.blank_line();
+
+    let mut imports = Vec::new();
+    let mut remainder = Vec::new();
+
+    for child in children {
+        match child.element().kind() {
+            SyntaxKind::ImportStatementNode => imports.push(child),
+            _ => remainder.push(child),
+        }
+    }
+
+    imports.sort_by(|a, b| {
+        let a = a
+            .element()
+            .as_node()
+            .expect("import statement node")
+            .as_import_statement()
+            .expect("import statement");
+        let b = b
+            .element()
+            .as_node()
+            .expect("import statement node")
+            .as_import_statement()
+            .expect("import statement");
+        let a_uri = a.uri().text().expect("import uri");
+        let b_uri = b.uri().text().expect("import uri");
+        a_uri.as_str().cmp(b_uri.as_str())
+    });
+
+    stream.blank_lines_allowed_between_comments();
+    for import in imports {
+        (&import).write(stream);
+    }
+
+    stream.blank_line();
+
+    for child in remainder {
+        (&child).write(stream);
+        stream.blank_line();
+    }
+}
+
+/// Formats a [`VersionStatement`](wdl_ast::VersionStatement).
+pub fn format_version_statement(element: &FormatElement, stream: &mut TokenStream<PreToken>) {
+    for child in element.children().expect("version statement children") {
+        (&child).write(stream);
+        stream.end_word();
+    }
+    stream.end_line();
+}
+
+/// Formats an [`InputSection`](wdl_ast::v1::InputSection).
+pub fn format_input_section(element: &FormatElement, stream: &mut TokenStream<PreToken>) {
+    let mut children = element.children().expect("input section children");
+
+    let input_keyword = children.next().expect("input section input keyword");
+    assert!(input_keyword.element().kind() == SyntaxKind::InputKeyword);
+    (&input_keyword).write(stream);
+    stream.end_word();
+
+    let open_brace = children.next().expect("input section open brace");
+    assert!(open_brace.element().kind() == SyntaxKind::OpenBrace);
+    (&open_brace).write(stream);
+    stream.increment_indent();
+
+    let mut inputs = Vec::new();
+    let mut close_brace = None;
+
+    for child in children {
+        match child.element().kind() {
+            SyntaxKind::BoundDeclNode | SyntaxKind::UnboundDeclNode => inputs.push(child),
+            SyntaxKind::CloseBrace => close_brace = Some(child),
+            _ => panic!("unexpected input section child"),
+        }
+    }
+
+    // TODO: sort inputs
```

Comment:
Should we knock this out in this PR? Or is that a future one?

Reply:
IMO it should wait till we have some configuration built-in. Don't have any data to back this claim up, but I think input sorting will be considered one of our most controversial features, and we probably want to provide an easy way to opt-out.